### PR TITLE
Refactor SDK Resolver into separate library, add simple workload manifest reader

### PR DESCRIPTION
**Pull request by dsplaisted**
Adds the following two projects:

- Microsoft.DotNet.SdkResolver
- Microsoft.DotNet.WorkloadManifestReader

The workload manifest format is super simple, to be replaced eventually by what is specified in https://github.com/dotnet/designs/pull/120.

Example of workload manifest:

```xml
<WorkloadManifest>
  <Workloads>
    <Workload Name="Xamarin.Android workload">
      <RequiredPack Name="Xamarin.Android.Workload"/>
    </Workload>
  </Workloads>
  <WorkloadPacks>
    <Pack Name="Xamarin.Android.Workload"
          Version="1.0.1"
          Kind="sdk" />
  </WorkloadPacks>
</WorkloadManifest>
```

### CODE REVIEW

**Review by wli3 — APPROVED** [review]

File: src/Resolvers/Microsoft.DotNet.WorkloadManifestReader/WorkloadManifest.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml.Linq;
+
+namespace Microsoft.NET.Sdk.WorkloadResolver
+{
+    class WorkloadManifest
+    {
+        public Dictionary<string, List<string>> Workloads { get; set; } = new Dictionary<string, List<string>>(StringComparer.OrdinalIgnoreCase);
```
Comment (wli3) [inline src/Resolvers/Microsoft.DotNet.WorkloadManifestReader/WorkloadManifest.cs:12]: The interface is the same as the real thing or a place holder too?

**Comment by wli3** [comment]
And for template pack, it will just have the pack "kind" be "template" right?

File: src/Resolvers/Microsoft.DotNet.WorkloadManifestReader/WorkloadManifest.cs (hunk 1 above)
Comment (dsplaisted) [reply src/Resolvers/Microsoft.DotNet.WorkloadManifestReader/WorkloadManifest.cs:12]: I'm sure the API will change.

**Comment by dsplaisted** [comment]
> And for template pack, it will just have the pack "kind" be "template" right?

I think so.  The API doesn't currently represent the pack "kind".

**Comment by wli3** [comment]
I'll just change the manifest reader in my PR later. Since nothing is final

**Comment by wli3** [comment]
Test failed

**Pull request merged**